client/client.go

```diff
@@ -103,6 +103,7 @@ func (c client) try(ctx context.Context, request Request, cancelFunc context.Can
 
 	for _, fm := range request.failManagers {
 		if err := fm.Check(resp); err != nil {
+			defer resp.Body.Close()
```

Comment:
we need to read the body to avoid the memory leak :)

Something like the following would do the trick I guess : 

```
if err := fm.Check(resp); err != nil {
    ioutil.ReadAll(resp.Body) // noop, we don't care about the error in case the body has already been read in the Check()
    resp.Body.Close()
    return Response{}, err
}
```

Reply:
:+1: